src/sparse_blas/backends/mklcpu/mklcpu_basic.cpp

```diff
@@ -0,0 +1,93 @@
+}
+
+sycl::event release_matrix_handle(sycl::queue &queue, detail::matrix_handle **handle,
+                                  const std::vector<sycl::event> &dependencies) {
+    return oneapi::mkl::sparse::release_matrix_handle(queue, detail::get_handle(handle),
+                                                      dependencies);
+}
+
+template <typename fpType, typename intType>
+std::enable_if_t<detail::are_fp_int_supported_v<fpType, intType>> set_csr_data(
+    sycl::queue &queue, detail::matrix_handle *handle, intType num_rows, intType num_cols,
+    index_base index, sycl::buffer<intType, 1> &row_ptr, sycl::buffer<intType, 1> &col_ind,
+    sycl::buffer<fpType, 1> &val) {
+    oneapi::mkl::sparse::set_csr_data(queue, detail::get_handle(handle), num_rows, num_cols, index,
+                                      row_ptr, col_ind, val);
+}
+
+template <typename fpType, typename intType>
+std::enable_if_t<detail::are_fp_int_supported_v<fpType, intType>, sycl::event> set_csr_data(
+    sycl::queue &queue, detail::matrix_handle *handle, intType num_rows, intType num_cols,
+    index_base index, intType *row_ptr, intType *col_ind, fpType *val,
+    const std::vector<sycl::event> &dependencies) {
+    return oneapi::mkl::sparse::set_csr_data(queue, detail::get_handle(handle), num_rows, num_cols,
+                                             index, row_ptr, col_ind, val, dependencies);
+}
+
+#define INSTANTIATE_SET_CSR_DATA(FP_TYPE, INT_TYPE)                                                \
+    template std::enable_if_t<detail::are_fp_int_supported_v<FP_TYPE, INT_TYPE>>                   \
+    set_csr_data<FP_TYPE, INT_TYPE>(                                                               \
+        sycl::queue & queue, detail::matrix_handle * handle, INT_TYPE num_rows, INT_TYPE num_cols, \
+        index_base index, sycl::buffer<INT_TYPE, 1> & row_ptr,                                     \
+        sycl::buffer<INT_TYPE, 1> & col_ind, sycl::buffer<FP_TYPE, 1> & val);                      \
+    template std::enable_if_t<detail::are_fp_int_supported_v<FP_TYPE, INT_TYPE>, sycl::event>      \
+    set_csr_data<FP_TYPE, INT_TYPE>(sycl::queue & queue, detail::matrix_handle * handle,           \
+                                    INT_TYPE num_rows, INT_TYPE num_cols, index_base index,        \
+                                    INT_TYPE * row_ptr, INT_TYPE * col_ind, FP_TYPE * val,         \
+                                    const std::vector<sycl::event> &dependencies);
+
+INSTANTIATE_SET_CSR_DATA(float, std::int32_t)
+INSTANTIATE_SET_CSR_DATA(double, std::int32_t)
+INSTANTIATE_SET_CSR_DATA(std::complex<float>, std::int32_t)
+INSTANTIATE_SET_CSR_DATA(std::complex<double>, std::int32_t)
+INSTANTIATE_SET_CSR_DATA(float, std::int64_t)
+INSTANTIATE_SET_CSR_DATA(double, std::int64_t)
+INSTANTIATE_SET_CSR_DATA(std::complex<float>, std::int64_t)
+INSTANTIATE_SET_CSR_DATA(std::complex<double>, std::int64_t)
```

Comment:
Please use semicolons at the end of the line where macros are called, and ensure that the macro does not end in a semi-colon.

```suggestion
                                    const std::vector<sycl::event> &dependencies)

INSTANTIATE_SET_CSR_DATA(float, std::int32_t);
INSTANTIATE_SET_CSR_DATA(double, std::int32_t);
INSTANTIATE_SET_CSR_DATA(std::complex<float>, std::int32_t);
INSTANTIATE_SET_CSR_DATA(std::complex<double>, std::int32_t);
INSTANTIATE_SET_CSR_DATA(float, std::int64_t);
INSTANTIATE_SET_CSR_DATA(double, std::int64_t);
INSTANTIATE_SET_CSR_DATA(std::complex<float>, std::int64_t);
INSTANTIATE_SET_CSR_DATA(std::complex<double>, std::int64_t);
```

Reply:
Done in https://github.com/oneapi-src/oneMKL/pull/374/commits/da857c7